lib/usd/hdMaya/adapters/meshAdapter.cpp

```diff
@@ -0,0 +1,444 @@
+    VtValue Get(const TfToken& key) override {
+        TF_DEBUG(HDMAYA_ADAPTER_GET)
+            .Msg(
+                "Called HdMayaMeshAdapter::Get(%s) - %s\n", key.GetText(),
+                GetDagPath().partialPathName().asChar());
+
+        if (key == HdTokens->points) {
+            MStatus status;
+            MFnMesh mesh(GetDagPath(), &status);
+            if (ARCH_UNLIKELY(!status)) { return {}; }
+            return GetPoints(mesh);
+        } else if (key == HdMayaAdapterTokens->st) {
+            return GetUVs();
+        }
+        return {};
+    }
+
+    size_t SamplePrimvar(
+        const TfToken& key, size_t maxSampleCount, float* times,
+        VtValue* samples) override {
+        if (maxSampleCount < 1) { return 0; }
+
+        if (key == HdTokens->points) {
+            MStatus status;
+            MFnMesh mesh(GetDagPath(), &status);
+            if (ARCH_UNLIKELY(!status)) { return 0; }
+            times[0] = 0.0f;
+            samples[0] = GetPoints(mesh);
+            if (maxSampleCount == 1 ||
+                !GetDelegate()->GetParams().enableMotionSamples) {
+                return 1;
+            }
+            times[1] = 1.0f;
+            MDGContextGuard guard(MAnimControl::currentTime() + 1.0);
+            samples[1] = GetPoints(mesh);
+            // FIXME: should we do this or in the render delegate?
+            return samples[1] == samples[0] ? 1 : 2;
+        } else if (key == HdMayaAdapterTokens->st) {
+            times[0] = 0.0f;
+            samples[0] = GetUVs();
+            return 1;
+        }
+        return 0;
+    }
+
+    HdMeshTopology GetMeshTopology() override {
+        MFnMesh mesh(GetDagPath());
+        const auto numPolygons = mesh.numPolygons();
+        VtIntArray faceVertexCounts;
+        faceVertexCounts.reserve(static_cast<size_t>(numPolygons));
+        VtIntArray faceVertexIndices;
+        faceVertexIndices.reserve(static_cast<size_t>(mesh.numFaceVertices()));
+        for (MItMeshPolygon pit(GetDagPath()); !pit.isDone(); pit.next()) {
+            const auto vc = pit.polygonVertexCount();
+            faceVertexCounts.push_back(vc);
+            for (auto i = decltype(vc){0}; i < vc; ++i) {
+                faceVertexIndices.push_back(pit.vertexIndex(i));
+            }
+        }
+
+        // TODO: Maybe we could use the flat shading of the display style?
+        return HdMeshTopology(
+#if MAYA_APP_VERSION >= 2019
+            (GetDelegate()->GetParams().displaySmoothMeshes ||
+             GetDisplayStyle().refineLevel > 0)
+                ? PxOsdOpenSubdivTokens->catmullClark
+                : PxOsdOpenSubdivTokens->none,
+#else
+            GetDelegate()->GetParams().displaySmoothMeshes
+                ? PxOsdOpenSubdivTokens->catmullClark
+                : PxOsdOpenSubdivTokens->none,
+#endif
+
+            UsdGeomTokens->rightHanded, faceVertexCounts, faceVertexIndices);
+    }
+
+    HdDisplayStyle GetDisplayStyle() override {
+#if MAYA_APP_VERSION >= 2019
```

Comment:
Any reason why this code path is disabled for Maya 2018?

Reply:
Just for completeness - this is the same issue discussed here:

https://github.com/Autodesk/maya-usd/pull/231#discussion_r374708276

I do wish I had a more concrete answer - ie, more details about what he was seeing, or ideally, some way to replicate his crash.  Leaving it does seem safest, though... and it will stop being an issue when we remove 2018 support!